### DISCUSSION TITLE
give specific component access option

### DISCUSSION THREAD

**Discussion by ghostdevv** [Budibase/budibase#1734] [Ideas]
I wanted my UI to have a table and then if you had admin rights then that table would have a edit button on each row, it would be great to be able to give components an access option so I could do that - even if it would only be available on things like containers

**Comment by joebudi** [comment]
Hey @ghostdevv 

If I understand correctly, I believe this is possible. I've created a quick video to demonstrate:
https://www.loom.com/share/4577fb54dd3c4f5eb50836f66f703266

Please let me know if that helps or if you have any further questions.

**Reply by ghostdevv** [reply]
Are you making two routes? One for user and one for admin? I am not quite sure I follow - could you export the budi app and send me a link so I could check it out?

**Reply by mike12345567** [reply]
Hi @ghostdevv - the method Joe is utilising is based on using two separate routes, and two entirely separate screens. The idea here is that if two screens are presented on the same route, but for different roles then when a user logs in they will be presented with the screen for their role, e.g. a screen on route `"/"` for an admin role, an admin will only be able to see, overriding say a screen on `"/"` for the basic role.

This is a little convoluted and confusing, in the future we will be working on the concept of "conditional UI" - the ability to setup certain components to only be shown when certain conditions are met.

**Reply by ghostdevv** [reply]
@mike12345567 wasn't aware you could do that, still not a fan would be great to be able to do the conditional ui type thing
